Review

sergiopetrovykh opened the pull request.

sergiopetrovykh committed:
12-1

sergiopetrovykh committed:
12-2

Переписаны тесты для cypress
Написаны Jest тесты для слайсов

mikebars1995 commented on cypress/e2e/constructor.cy.ts:
```diff
@@ -0,0 +1,115 @@
+
+  describe('Проверка модальных окон', function () {
+    it('Открытие модального окна', function () {
+      cy.get('[data-cy="modal"]').should('not.exist');
```
**Нужно исправить**

Если **строка-селектор** повторяется более 2х раз в коде, значит, этот селектор нужно выносить в **константу**, чтобы можно было одним движением изменить его и не искать по всему коду дублирования этого селектора. 

**Можно лучше**

* Лучше использовать переменные **alias** в тестах, чтобы 1 раз найти элемент и обращаться к нему. Подробнее тут https://docs.cypress.io/api/commands/as#DOM-element

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)


Работа проделана огромная:


* Все юнит-тесты (33 шт) проходят успешно
* Все cypress-тесты проходят успешно
* Код тестов аккуратный и понятный
* Отлично, что базовый урл вынесен в конфиг


но есть некоторые недочеты:

* Если **строка-селектор** повторяется более 2х раз в коде, значит, этот селектор нужно выносить в **константу**, чтобы можно было одним движением изменить его и не искать по всему коду дублирования этого селектора.

sergiopetrovykh committed:
12-3

Внесены правки после ревью:
Исправлен cypress-тест: Если строка-селектор повторяется более 2х раз в коде, значит, этот селектор нужно выносить в константу, чтобы можно было одним движением изменить его и не искать по всему коду дублирования этого селектора

mikebars1995 reviewed (commented):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
